### ISSUE TITLE
Normalize the value function Q(s, a) from the neural network to [0,1)

### ISSUE THREAD

**Issue by Chicoryn** [Chicoryn/dream-go#7]
The value function _Q(s, a)_ in [PUCT](https://www.nature.com/articles/nature24270) or [RAVE](http://www.cs.utexas.edu/~pstone/Courses/394Rspring13/resources/mcrave.pdf) formula is the probability of winning for a given state and action.

In the current implementation we use the output from the neural network (which is between -1 and +1) as the value function, but this might give the wrong result. We should try normalizing the result of the value head _v_ to see if following the formula gives better results.

We might need to re-tune the hyper-parameters of the monte-carlo search after this change.

**Comment by Chicoryn** [comment]
Fixing this is simple, we just need to change _v_ to _v / 2.0 + 0.5_ in the `mcts::forward` function.

## Sanity checks (vs GNU Go)
To make sure we did not do something stupid we ran a tournament against _GNU Go_ using the adjusted value function. The final score was `3 - 0` in favour of Dream Go which suggests that the search is still working.

[dg-vs-gnugo.zip](https://github.com/Chicoryn/dream-go/files/1592411/dg-vs-gnugo.zip)

## Parameter Tuning

### RAVE
```
TODO
```

### PUCT
```
TODO
```

## Versus previous version
We ran a tournament between the following engines to determine how the change in the value function affected the strength of the program.

* `dg-323293f` (commit `323293f`)
* `dg-323293f-puct` (commit `323293f` with normalized value function, using `puct`)
* `dg-323293f-rave` (commit `323293f` with normalized value function, using `rave`)

```
TODO
```

**Comment by Chicoryn** [comment]
This seems to be an pretty clear improvement over previous versions. So I am going to merge this change with PUCT as the algorithm since there is not enough data to determine a clear winner between PUCT and RAVE.
